code/modules/projectiles/projectile.dm

```diff
@@ -258,6 +276,17 @@
 		error = dist_x/2 - dist_y
 	else
 		error = dist_y/2 - dist_x
+
+	target_angle = round(Get_Angle(starting,target))
+
+	if(linear_movement)
+		//If the icon has not been added yet
+		if( !("[icon_state][target_angle]" in bullet_master) )
+			var/icon/I = new(icon,"[icon_state]_pixel") //Generate it.
+			I.Turn(target_angle+45)

```

Comment:
Well it had nothing to do with whether I used Turn or transform, I realized that the beam would appear borked either way, for one simple reason. You see, this PR adds linear beams and projectile paths ( https://dl.dropboxusercontent.com/u/33846352/directional.webm ) but you can enable or disable this linearity per (bullet) projectile path. When I started working on the PR, whether or not a path would use linearity was decided by whether its animate_movement var was 0 or not. However I added a linear_movement var to replace that (in case someone wanted a non-animate movement projectile that doesn't follow a linear path). HOWEVER I forgot to set linear_movement to 0 for beams (which are now linear to begin with) so the transformation would somehow get applied twice.